R/rules.R

```diff
@@ -169,121 +177,150 @@ sv_regex <- function(pattern, message, ignore.case = FALSE, perl = FALSE,
-#'   the rule.
-#' @param allowMultiple If `FALSE` (the default), then the length of the input
+#' @param message The validation error message to use if a value is not numeric.
+#' @param allow_multiple If `FALSE` (the default), then the length of the input
 #'   vector must be exactly one; if `TRUE`, then any length is allowed
 #'   (including a length of zero; use [sv_required()] if one or more values
 #'   should be required).
-#' @param allowNA If `FALSE` (the default), then any `NA` element will cause
-#'   validation to fail.
-#' @param allowNaN If `FALSE` (the default), then any `NaN` element will cause
-#'   validation to fail.
-#' @param allowInfinite If `FALSE` (the default), then any `Inf` or `-Inf`
-#'   element will cause validation to fail.
+#' @param allow_na,allow_nan If `FALSE` (the default for both options), then any
+#'   `NA` or `NaN` element will cause validation to fail.
+#' @param allow_inf If `FALSE` (the default), then any `Inf` or `-Inf` element
+#'   will cause validation to fail.
+#' 
 #' @return A function suitable for using as an
 #'   [`InputValidator$add_rule()`][InputValidator] rule.
 #'
 #' @examples
 #' # Ignore withReactiveDomain(), it's just required to get this example to run
 #' # outside of Shiny
 #' shiny::withReactiveDomain(shiny::MockShinySession$new(), {
-#' 
+#'
 #'   iv <- InputValidator$new()
-#' 
-#'   iv$add_rule("count", sv_integer())
-#'   iv$add_rule("count", ~if (. <= 0) "A positive value is required")
 #'
+#'   iv$add_rule("count", sv_numeric())
+#'   iv$add_rule("count", ~if (. <= 0) "A positive value is required")
 #' })
+#'
 #' @export
-sv_numeric <- function(message = "A number is required", allowMultiple = FALSE,
-  allowNA = FALSE, allowNaN = FALSE, allowInfinite = FALSE) {
-
+sv_numeric <- function(message = "A number is required",
+                       allow_multiple = FALSE,
+                       allow_na = FALSE,
+                       allow_nan = FALSE,
+                       allow_inf = FALSE) {
   force(message)
-  force(allowMultiple)
-  force(allowNA)
-  force(allowNaN)
-  force(allowInfinite)
+  force(allow_multiple)
+  force(allow_na)
+  force(allow_nan)
+  force(allow_inf)
 
-  function(value) {
-    if (!is.numeric(value)) {
-      return(message)
-    }
-    if (length(value) == 0) {
-      return(message)
-    }
-    if (!allowMultiple && length(value) != 1) {
-      return(message)
-    }
-    if (!allowNA && any(is.na(value))) {
-      return(message)
-    }
-    if (!allowNaN && any(is.nan(value))) {
-      return(message)
-    }
-    if (!allowInfinite && any(!is.finite(value))) {
-      return(message)
+  compose_rules(
+    sv_basic(
+      allow_multiple = allow_multiple,
+      allow_na = allow_na,
+      allow_nan = allow_nan,
+      allow_inf = allow_inf
+    ),
+    function(value) {
+      
+      if (length(value) == 0) {
+        return(err_condition_messages[["err_zero_length_value"]])
```

Comment:
I'm worried that incorrect keys (`"err_zero_length_val"`) would result in silent `NULL` result (i.e. the check passes). Two suggestions:

1. Don't use an `err_condition_messages` list, just have variables with a common prefix--if you misspell, it'll just be a `x not found` error
2. Make `err_condition_messages` a function that notices if the key is not known and throws.

Reply:
I chose option 1 and it's been committed.